packages/commonwealth/server/routes/spam/markCommentAsSpam.ts

```diff
@@ -0,0 +1,69 @@
+import { AppError } from 'common-common/src/errors';
+import type { NextFunction, Request, Response } from 'express';
+import type { DB } from '../../models';
+import { Op, Sequelize } from 'sequelize';
+import { success } from '../../types';
+import { findAllRoles } from '../../util/roles';
+
+export const Errors = {
+  InvalidCommentId: 'Comment ID invalid',
+  NotLoggedIn: 'Not logged in',
+  CommentNotFound: 'Could not find Comment',
+  NotAdmin: 'Not an admin',
+};
+
+export default async (
+  models: DB,
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  const commentId = req.params.id;
+  if (!commentId) {
+    return next(new AppError(Errors.InvalidCommentId));
+  }
+
+  if (!req.user) {
+    return next(new AppError(Errors.NotLoggedIn));
+  }
+
+  const comment = await models.Comment.findOne({
+    where: {
+      id: commentId,
+    },
+  });
+  if (!comment) {
+    return next(new AppError(Errors.CommentNotFound));
+  }
+  const userOwnedAddressIds = (await req.user.getAddresses())
```

Comment:
Wondering if there's a helper that exists for this check of is author or admin/mod, which is pretty common.